packages/devextreme/testing/tests/DevExpress.ui.widgets/menu.tests.js

```diff
@@ -1258,6 +1402,97 @@ QUnit.module('Menu tests', {
+
+        const nestedSubmenu1 = $subItem1.find(`.${DX_SUBMENU_CLASS}`)[0];
+
+        assert.strictEqual(getDomNode(handlerShowing.args[1][0].rootItem), $rootItem[0], '2nd level show');
+        assert.strictEqual(handlerShowing.args[1][0].itemData.id, 11);
+        assert.strictEqual(getDomNode(handlerShowing.args[1][0].submenuContainer), nestedSubmenu1);
+        assert.strictEqual(getDomNode(handlerShown.args[1][0].rootItem), $rootItem[0]);
+        assert.strictEqual(handlerShown.args[1][0].itemData.id, 11);
+        assert.strictEqual(getDomNode(handlerShown.args[1][0].submenuContainer), nestedSubmenu1);
+
+        $($subItem2).trigger('dxclick');
+
+        const nestedSubmenu2 = $subItem2.find(`.${DX_SUBMENU_CLASS}`)[0];
+
+        assert.strictEqual(getDomNode(handlerHiding.args[0][0].rootItem), $rootItem[0], '2nd level hide');
+        assert.strictEqual(handlerHiding.args[0][0].itemData.id, 11);
+        assert.strictEqual(getDomNode(handlerHiding.args[0][0].submenuContainer), nestedSubmenu1);
+        assert.strictEqual(getDomNode(handlerHidden.args[0][0].rootItem), $rootItem[0]);
+        assert.strictEqual(handlerHidden.args[0][0].itemData.id, 11);
+        assert.strictEqual(getDomNode(handlerHidden.args[0][0].submenuContainer), nestedSubmenu1);
+
+        assert.strictEqual(getDomNode(handlerShowing.args[2][0].rootItem), $rootItem[0], 'open 2nd level again');
+        assert.strictEqual(handlerShowing.args[2][0].itemData.id, 12);
+        assert.strictEqual(getDomNode(handlerShowing.args[2][0].submenuContainer), nestedSubmenu2);
+        assert.strictEqual(getDomNode(handlerShown.args[2][0].rootItem), $rootItem[0]);
+        assert.strictEqual(handlerShown.args[2][0].itemData.id, 12);
+        assert.strictEqual(getDomNode(handlerShown.args[2][0].submenuContainer), nestedSubmenu2);
+
+        $(document).trigger('dxpointerdown');
+
+        assert.strictEqual(getDomNode(handlerHiding.args[1][0].rootItem), $rootItem[0], '1st level hide');
+        assert.strictEqual(handlerHiding.args[1][0].itemData.id, 1);
+        assert.strictEqual(getDomNode(handlerHiding.args[1][0].submenuContainer), rootSubmenu);
+        assert.strictEqual(getDomNode(handlerHidden.args[2][0].rootItem), $rootItem[0]);
+        assert.strictEqual(handlerHidden.args[2][0].itemData.id, 1);
+        assert.strictEqual(getDomNode(handlerHidden.args[2][0].submenuContainer), rootSubmenu);
+
+        assert.strictEqual(getDomNode(handlerHiding.args[2][0].rootItem), $rootItem[0], '2nd level hide');
+        assert.strictEqual(handlerHiding.args[2][0].itemData.id, 12);
+        assert.strictEqual(getDomNode(handlerHiding.args[2][0].submenuContainer), nestedSubmenu2);
+        assert.strictEqual(getDomNode(handlerHidden.args[1][0].rootItem), $rootItem[0]);
+        assert.strictEqual(handlerHidden.args[1][0].itemData.id, 12);
+        assert.strictEqual(getDomNode(handlerHidden.args[1][0].submenuContainer), nestedSubmenu2);
+    });
+
```

Comment:
Please add a test for scenario for which we extend this API. Like `it should be possible to restrict specific submenu height using API`